File: mascii2/src/main/java/com/kastkode/mascii2/MasciiParseErrorListener.java
package com.kastkode.mascii2;
import java.util.LinkedList;
import java.util.List;

import org.antlr.v4.runtime.BaseErrorListener;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.Recognizer;

public class MasciiParseErrorListener extends BaseErrorListener {

	List<String> msgs = new LinkedList<>();
	
	@Override
	public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine,
			String msg, RecognitionException e) {
		super.syntaxError(recognizer, offendingSymbol, line, charPositionInLine, msg, e);
		msgs.add(String.format("%s at %s:%s %s", offendingSymbol, line, charPositionInLine, msg));
	}

	public List<String> getMsgs() {
		return msgs;
	}

	public void setMsgs(List<String> msgs) {
		this.msgs = msgs;
	}
	
	

}

File: mascii2-typescript/src/MasciiParseErrorListener.ts
import { ErrorListener, RecognitionException, Recognizer } from 'antlr4';

export class MasciiParseErrorListener extends ErrorListener<unknown> {
    private msgs: string[] = [];

    override syntaxError(
        recognizer: Recognizer<unknown>,
        offendingSymbol: unknown,
        line: number,
        charPositionInLine: number,
        msg: string,
        _e: RecognitionException | undefined,
    ): void {
        this.msgs.push(`${offendingSymbol} at ${line}:${charPositionInLine} ${msg}`);
    }

    getMsgs(): string[] {
        return this.msgs;
    }
}
